consensus/hotstuff/signature/randombeacon_signer_store.go

```diff
@@ -26,59 +27,55 @@ func NewEpochAwareRandomBeaconKeyStore(epochLookup module.EpochLookup, keys stor
 // It returns:
 //   - (signer, nil) if DKG succeeded locally in the epoch of the view, signer is not nil
-//   - (nil, protocol.ErrNextEpochNotCommitted) if no epoch found for given view
-//   - (nil, DKGFailError) if DKG failed locally in the epoch of the view
+//   - (nil, model.ErrViewForUnknownEpoch) if no epoch found for given view
+//   - (nil, module.ErrNoBeaconKeyForEpoch) if beacon key for epoch is unavailable
 //   - (nil, error) if there is any exception
 func (s *EpochAwareRandomBeaconKeyStore) ByView(view uint64) (crypto.PrivateKey, error) {
-	// fetching the epoch by view, if epoch is found, then DKG must have been completed
 	epoch, err := s.epochLookup.EpochForViewWithFallback(view)
 	if err != nil {
 		return nil, fmt.Errorf("could not get epoch by view %v: %w", view, err)
 	}
 
-	// when DKG has completed,
-	// 1. if a node successfully generated the DKG key, the valid private key will be stored in database.
-	// 2. if a node failed to generate the DKG key, we will save a record in database to indicate this
-	//       node has no private key for this epoch
-	// within the epoch, we can lookup my random beacon private key for the epoch. There are 3 cases:
-	// 1. DKG has completed, and the private key is stored in database, and we can retrieve it (happy path)
-	// 2. DKG has completed, but we failed it, and we marked in the database
+	// When DKG has completed,
+	//   1. if a node successfully generated the DKG key, the valid private key will be stored in database.
+	//   2. if a node failed to generate the DKG key, we will save a record in database to indicate this
+	//      node has no private key for this epoch.
+	// Within the epoch, we can look up my random beacon private key for the epoch. There are 4 cases:
+	//   1. DKG has completed, and the private key is stored in database, and we can retrieve it (happy path)
+	//   2. DKG has completed, but we failed it, and we marked in the database
 	// 		that there is no private key for this epoch (unhappy path)
-	// 3. DKG has completed, but for some reason we don't find the private key in the database (exception)
-	// 4. DKG was not completed (fatal error, we should not run into with EECC, because we still stay at
-	// 		the current Epoch where DKG has completed.
+	//   3. DKG has completed, but for some reason we don't find the private key in the database (exception)
+	//   4. DKG was not completed (exception, results in EECC)
 	key, found := s.privateKeys[epoch]
 	if found {
-		// A nil key means that we don't have a Random Beacon key for this epoch.
+		// a nil key means that we don't (and will never) have a beacon key for this epoch
 		if key == nil {
-			return nil, fmt.Errorf("DKG for epoch %v failed, at view %v: %w",
-				epoch, view, module.DKGFailError)
+			return nil, fmt.Errorf("beacon key for epoch %d (queried view: %d) never available: %w", epoch, view, module.ErrNoBeaconKeyForEpoch)
 		}
 		return key, nil
```

Comment:
```suggestion
		// case 1
		return key, nil
```

Reply:
Added comments noting each case in https://github.com/onflow/flow-go/pull/3950/commits/235ec9464e614232c370ec8830133d391898c6a0